### ISSUE TITLE
Tests should not rely on go-libp2p-kad-dht

### ISSUE THREAD

**Issue by hsanjuan** [ipfs/go-namesys#5]
This is a follow up issue to https://github.com/ipfs/go-ipfs/pull/7925.

@aschmahmann in the tests, we use libp2p nodes with dht. This tests ipns publishing more or less realistically. What should we using instead?

**Comment by Stebalien** [comment]
Unfortunately, it's a bit more than that. We _also_ specify the "dht record count". However, I _think_ we can likely just drop that and replace it with a better option.

At this point, we should have simple "allow-stale" flag (or something like that):

* When set, we take the first record found and return it.
* When unset, we wait until the query completes.

**Comment by hacdias** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
